open_dread_rando/files/custom_scenario.lua

```diff
@@ -48,15 +48,35 @@ function Scenario.InitScenario(arg1, arg2, arg3, arg4)
     init_scenario(arg1, arg2, arg3, arg4)
 
     local playerSection =  Game.GetPlayerBlackboardSectionName()
+    local currentSaveRandoHash = Blackboard.GetProp(playerSection, "THIS_RANDO_HASH")
     local randoInitialized = Blackboard.GetProp(playerSection, "RANDO_GAME_INITIALIZED")
 
+    Game.LogWarn(0, ("Cross-checking seed hashes. The current patch's hash is %q, and the current save's hash is %q."):format(Init.sThisRandoSeedHash, tostring(currentSaveRandoHash)))
+
+    -- Cross-check the seed hash in the Blackboard with the one in Init.sThisRandoSeedHash to make sure they match.
+    -- If they don't, show a warning to the player, and DO NOT save over their game!
+    if currentSaveRandoHash ~= Init.sThisRandoSeedHash then
+        Game.AddSF(0.8, Scenario.ShowNotRandoGameMessage, "")
+        return
+    end
+
     if not randoInitialized then
         Blackboard.SetProp(playerSection, "RANDO_GAME_INITIALIZED", "b", true)
         Game.AddSF(0.9, Init.SaveGameAtStartingLocation, "")
         Game.AddSF(0.8, Scenario.ShowText, "")
     end
 end
 
+local warning_messages_seen = 0
+local total_warning_messages = 2
```

Comment:
Could be similar to the starting items message: the error message is set via an additional field in the config, so we can have a dynamic number of messages.

Reply:
don't worry about this for now; I have a refactor on my objectives branch to generalize this system anyway so I'll clean this up when I resolve merge conflicts